src/pytest_servers/gcs.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import time
+
+import pytest
+import requests
+
+from .utils import get_free_port
+
+logger = logging.getLogger(__name__)
+
+GCS_DEFAULT_PORT = 4443
+
+
+@pytest.fixture(scope="session")
+def fake_gcs_server(docker_client):
+    if docker_client is None:
+        logger.warning(
+            "fake-gcs-server cannot run because docker is not available."
+        )
+        yield None
```

Comment:
This gets tricky. Since `tmp_upath_factory` uses different fixtures (`s3`, `azurite`, `fake_gc_server`, ...), if we use `pytest.skip` here, the whole `tmp_upath_factory` will fixture be skipped, skipping tests that could have been run (e.g. `s3`).

So, since not all fixtures used by `tmp_upath_factory` are/will be dependent on docker, I'd rather keep it like this for now, and use `raise_or_skip` in `mktemp`. If in the future we end up with all fixtures dependent on docker, we can just replace this with `pytest.skip`.